flytekit/common/tasks/sdk_runnable.py

```diff
@@ -25,12 +25,53 @@
 from flytekit.configuration import internal as _internal_config
 from flytekit.configuration import resources as _resource_config
 from flytekit.configuration import sdk as _sdk_config
+from flytekit.configuration import secrets
 from flytekit.engines import loader as _engine_loader
 from flytekit.interfaces.stats import taggable
 from flytekit.models import literals as _literal_models
 from flytekit.models import task as _task_models
 
 
+class SecretsManager(object):
+    """
+    This provides a secrets resolution logic at runtime.
+    The resolution order is
+      - Try env var first. The env var should have the configuration.SECRETS_ENV_PREFIX. The env var will be all upper
+         cased
+      - If not then try the file where the name matches the key as lower case in configuration.SECRETS_DEFAULT_DIR
+
+    All configuration values can always be overriden by injecting an environment variable
+    """
+
+    def __init__(self):
+        self._base_dir = str(secrets.SECRETS_DEFAULT_DIR.get()).strip()
+        self._file_prefix = str(secrets.SECRETS_FILE_PREFIX.get()).strip()
+        self._env_prefix = str(secrets.SECRETS_ENV_PREFIX.get()).strip()
+
+    def get(self, secrets_key: str) -> str:
+        """
+        Retrieves a secret using the resolution order -> Env followed by file. If not found raises a ValueError
+        """
+        if secrets_key is None or secrets_key == "":
+            raise ValueError("Bad secrets key. Cannot be an empty string or None.")
+
+        env_var = self.get_secrets_env_var(secrets_key)
+        fpath = self.get_secrets_file(secrets_key)
+        v = os.environ.get(env_var)
+        if v is not None:
+            return v
+        if os.path.exists(fpath):
+            with open(fpath, "r") as f:
+                return f.read()
```

Comment:
maybe a `strip()` here in case of newlines??

Reply:
will do

Reply:
done